streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamImplTest.java

```diff
@@ -1865,6 +1865,17 @@ public void shouldNotAllowNullNamedOnTransformWithStoreName() {
         assertThat(exception.getMessage(), equalTo("named can't be null"));
     }
 
+    @Test
+    public void shouldNotAllowBadTransformerSupplierOnFlatTransform() {
```

Comment:
It might be nice to replicate this test for all "siblings" of `flatTransform` (including the "Value" variants). -- Also note that there is `KStream#process()` and `StreamsBuilder#addGlobalStateStore), too, that should also be tested.

Reply:
The "Value" variants use a `ValueTransformerSupplier` instead of a `TransformerSupplier`. Does it makes sense to extend the supplier usage verification to `ValueTransformerSupplier` as well?